### PR TITLE
update CHANGELOG with current unreleased changes

### PR DESCRIPTION
**Pull request by zachschuermann**
added current unreleased changes before `0.5.0` release (separate PR with hopefully a bit of automation)

view [rendered](https://github.com/zachschuermann/delta-kernel-rs/blob/d2211ff9158f9c8f185d453d818f4bacda8fec55/CHANGELOG.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,125 @@
...
+   `ColumnName::parse_column_name_list()` [\#458]
+9. New `VacuumProtocolCheck` table feature [\#454]
+10. Added `Cdc` action support [\#506]
+11. (early CDF read support) New `TableChanges` type to read CDF from a table between versions
+    [\#505]
+12. (early CDF read support) Builder for scans on `TableChanges` [\#521]
+13. New `TableProperties` struct which can parse tables' `metadata.configuration` [\#453] [\#536]
+
+**Implemented enhancements:**
+- FFI examples now use AddressSanitizer [\#447]
```
Comment (OussamaSaoudi-db) [inline CHANGELOG.md:53]: Nit: How come we use `-` for these bullet points, but numbers in the Additions?

Comment (zachschuermann) [reply CHANGELOG.md:53]: no reason lol other than it felt nice to have structure in the changes/additions and felt less useful here. want numbers everwhere?

Comment (OussamaSaoudi-db) [reply CHANGELOG.md:53]: I have a slight preference towards a consistent bullets, but I'm cool with either.

**Comment by OussamaSaoudi-db** [comment]
Besides addressing the TODO on line 14 of Breaking changes, it LGTM.

**Review by OussamaSaoudi-db — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,126 @@
...
+*Breaking*
+
+1. `Expression::Column(String)` is now `Expression::Column(ColumnName)` [\#400]
+2. delta_kernel_ffi::expressions moved into two modules:
+   `delta_kernel_ffi::expressions::engine` and `delta_kernel_ffi::expressions::kernel` [\#363]
+3. FFI: removed (hazardous) `impl From` for `KernelStringSlize` and added `unsafe` constructor
+   instead [\#441]
+4. Moved `LogSegment` into its own module (`log_segment::LogSegment`) [\#438]
+5. `DvInfo` now implements `Clone`, `PartialEq`, and `Eq` [\#468]
+6. `Stats` now implements `Debug`, `Clone`, `PartialEq`, and `Eq` [\#468]
```
Comment (scovich) [inline CHANGELOG.md:18]: These are not breaking changes? At least, our PR template states:
> Note that _new_ public APIs are not considered breaking.

**Review by scovich — APPROVED** [review]
LGTM except two additions masquerading as breaking changes.

**Review by nicklan — APPROVED** [review]
thanks! wow that's a lot of changes

File: CHANGELOG.md (hunk 2 above)
Comment (zachschuermann) [reply CHANGELOG.md:18]: ah good catch thanks!

**Pull request merged**